WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java

```diff
@@ -93,22 +92,25 @@ public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError e
     }
 
     @Override
-    public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+    public WebResourceResponse shouldInterceptRequest(WebView view, String stringUrl) {
         // Intercept requests for private images and add the WP.com authorization header
-        if (mBlog != null && mBlog.isPrivate() && !TextUtils.isEmpty(mToken) && UrlUtils.isImageUrl(url)) {
-            DefaultHttpClient client = new DefaultHttpClient();
-            HttpGet httpGet = new HttpGet(url);
-            httpGet.setHeader("Authorization", "Bearer " + mToken);
+        if (mBlog != null && mBlog.isPrivate() && !TextUtils.isEmpty(mToken) && UrlUtils.isImageUrl(stringUrl)) {
             try {
-                HttpResponse httpResponse = client.execute(httpGet);
-                InputStream responseInputStream = httpResponse.getEntity().getContent();
-                return new WebResourceResponse(httpResponse.getEntity().getContentType().toString(),
-                        "UTF-8", responseInputStream);
+                URL url = new URL(stringUrl);
+                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection.setRequestProperty("Authorization", "Bearer " + mToken);
```

Comment:
I think the timeouts default to zero (no timeout) which may [have repercussions](https://code.google.com/p/android/issues/detail?id=40874#c16)